### ISSUE TITLE
Upgrade dependency on oauth2 to 1.6?

### ISSUE THREAD

**Issue by cthorner** [Shopify/omniauth-shopify-oauth2#90]
At the moment this gem is dependent on omniauth-oauth2 1.5, but a new version has been out since 2018(v1.6). Can we update the dependency to allow 1.6?

https://github.com/Shopify/omniauth-shopify-oauth2/blob/2359fe8229add27afb88eeebbe7306a8e4ed1451/omniauth-shopify-oauth2.gemspec#L20

**Comment by andyw8** [comment]
We can look into this, but is there a particular feature/fix from 1.6 that you need?

omniauth-oauth2 doesn't seem to provide a changelog so makes it little difficult to determine what's changed.

**Comment by erichmachado** [comment]
Hi @andyw8, we're having an issue trying to install the latest version of `omniauth-shopify-oauth2` on a project that uses `omniauth-google-oauth2` and it's related to that dependency restriction:
```
Bundler could not find compatible versions for gem "omniauth-oauth2":
  In snapshot (Gemfile.lock):
    omniauth-oauth2 (= 1.7.0)
  In Gemfile:
    omniauth-google-oauth2 was resolved to 0.8.0, which depends on
      omniauth-oauth2 (>= 1.6)
    omniauth-shopify-oauth2 (~> 2.2) was resolved to 2.2.3, which depends on
      omniauth-oauth2 (~> 1.5.0)
Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```
Would it be possible to change that in order to enable using `omniauth-shopify-oauth2` with newer versions of `omniauth-oauth2`?

The changelogs for the last two releases are apparently available on those links:
1. https://github.com/omniauth/omniauth-oauth2/releases/tag/v1.6.0
2. https://github.com/omniauth/omniauth-oauth2/releases/tag/v1.7.0
